src/api-next/api-next.ts

```diff
@@ -1601,4 +1602,50 @@ export class IBApiNext {
         .pipe(map((v: { all: DepthMktDataDescription[] }) => v.all))
     );
   }
+
+  /** nextValidId event handler */
+  private readonly onNextValidId = (
+    subscriptions: Map<number, IBApiNextSubscription<number>>,
+    orderId: number
+  ): void => {
+    subscriptions.forEach((sub) => {
+      sub.next({
+        all: orderId,
+      });
+      sub.complete();
+    });
+  };
+
+  /**
+   * Requests the next valid order ID at the current moment.
+   */
+  getNextValidOrderId(): Promise<number> {
+    return lastValueFrom(
+      this.subscriptions
+        .register<number>(
+          () => {
+            this.api.reqIds();
+          },
+          undefined,
+          [[EventName.nextValidId, this.onNextValidId]],
+          "getNextUnusedOrderId" // use same instance id each time, to make sure there is only 1 pending request at time
+        )
+        .pipe(map((v: { all: number }) => v.all))
+    );
+  }
+
+  /**
+   * Places or modifies an order.
+   *
+   * @param id The order's unique identifier.
+   * Use a sequential id starting with the id received from the getNextValidOrderId() method.
+   * If a new order is placed with an order ID less than or equal to the order ID of a previous order an error will occur.
+   * @param contract The order's [[Contract]].
+   * @param order The [[Order]] object.
+   *
+   * @see [[reqAllOpenOrders]], [[reqAutoOpenOrders]], [[reqOpenOrders]], [[cancelOrder]], [[reqGlobalCancel]]
+   */
+  placeOrder(id: number, contract: Contract, order: Order): void {
+    this.api.placeOrder(id, contract, order);
```

Comment:
Can we wrap it somehow, so it will automatically get the newOrderId? Or is there some kind of specific use case we can't do so?

@mfrener

Reply:
Did this initially, until read doc again :D 
" Places or **modifies** an order." It will always place a new order, if we allocated the order id. So this must be left user, as only he can decided if want to use a new order id, or modify an existing one.

Reply:
hmmm or maybe we do a
`placeNewOrder(contract: Contract, order: Order): number`
and
`modifyOrder(id: number, contract: Contract, order: Order): number`
?
derives from what IB does, but IMHO usably is better. @Tsopic what do you think?

Reply:
Seems huge usability improvement, don't even need to check the docs to know what it does.